### PR TITLE
INF-807 Added multi_sink.py to support sending data to several sinks from stdin

### PR DESCRIPTION
**Pull request by JavierMF**

### CODE REVIEW

File: sinks/multi_sink.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+
+class Sinks:
+
+    def __init__(self, sink_commands):
+        self.sinks = []
+        self.lines_sent = 0
+        for param in sink_commands:
+            print(f'Executing: {param}')
+            self.sinks.append(subprocess.Popen(param.split(), stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.STDOUT))
```
Comment (dpaneda) [inline sinks/multi_sink.py:14]: I think that maybe we should protect this code if the Popen fails to create the process. I think is acceptable that one of the sink dies at some point making both sinks to fail, but we should at least be able to ignore the sinks that fail to be launched

**Review by dpaneda — APPROVED** [review]

**Review by own3dh2so4 — APPROVED** [review]

File: sinks/multi_sink.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    def write_lines(self, lines):
+        self.lines_sent += len(lines)
+        encoded_lines = [line.encode('utf-8') for line in lines]
+        [sink.stdin.writelines(encoded_lines) for sink in self.sinks]
+
+
+sinks = Sinks(sys.argv[1:])
```
Comment (tsotnikov) [inline sinks/multi_sink.py:30]: Can you put an example of how it is supposed to be configured?

This is our current configuration: https://github.com/Telefonica/dockerfiles/blob/master/statsite/statsite.conf#L15

Comment (JavierMF) [reply sinks/multi_sink.py:30]: I'm not sure if you ask to put the example here or in the code. I'll put it here and you tell me.
In an scenario where we want to send to different ports in the same server, we would use something like:

`python multi_sink.py "python /sinks/graphite.py GRAPHITE_HOST 2004 'stats.' 3 'pickle'" "python /sinks/graphite.py GRAPHITE_HOST 2005 'stats.' 3 'pickle'"`

File: sinks/multi_sink.py (hunk 1 above)
Comment (JavierMF) [reply sinks/multi_sink.py:14]: Popen does not fail if the command can not be executed (tried with wrong sink name and with a sink raising exception) but error was not visible. I added code to show sink output if any. Now the output is like this:

Executing 1: python sink_a.py
Executing 2: python sink_b.py
Error in 1: b'Traceback (most recent call last):\n  File "sink_a.py", line 3, in <module>\n    raise Exception()\nException\n'
10 lines sent

File: sinks/multi_sink.py [hunk 3]
```diff
@@ -9,15 +9,21 @@ class Sinks:
...
             self.sinks.append(subprocess.Popen(param.split(), stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.STDOUT))
 
     def write_lines(self, lines):
         self.lines_sent += len(lines)
         encoded_lines = [line.encode('utf-8') for line in lines]
         [sink.stdin.writelines(encoded_lines) for sink in self.sinks]
 
+    def print_errors(self):
+        for idx, sink in enumerate(self.sinks):
+            (out, _) = sink.communicate()
```
Comment (tsotnikov) [inline sinks/multi_sink.py]: https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate

communicate returns (stdout, stderr), we want to print them both.

**Review by tsotnikov — APPROVED** [review]

File: sinks/multi_sink.py (hunk 3 above)
Comment (JavierMF) [reply sinks/multi_sink.py]: Ok. I'll add to print stderr too. For my tests I saw error messages were received in stdout but maybe some other sink writes something to stderr.

**Pull request merged**